src/global/tools/TestSentenceUtil.java

```diff
@@ -13,6 +16,43 @@ public static boolean hasRandom(Clause[] testSentence) {
         return false;
     }
 
+    public static Clause[] expandTestSentence(Clause[] testSentence, Clause[][] multipleInjections){
+        int multipleInjectionIndex = 0;
+        ArrayList<Clause> modifiedTestSentence = new ArrayList<>();
+        for (Clause currClause : testSentence) {
+            if (!(currClause instanceof PlaceHolder)) {
+                modifiedTestSentence.add(currClause);
+                continue;
+            }
+
+            Clause[] clausesToInject = multipleInjections[multipleInjectionIndex];
+            modifiedTestSentence.addAll(Arrays.asList(clausesToInject));
+            multipleInjectionIndex++;
+        }
+
+        Clause[] outputClauses = new Clause[modifiedTestSentence.size()];
+        for(int i = 0; i < modifiedTestSentence.size(); i++) {
+            outputClauses[i] = modifiedTestSentence.get(i);
+        }
```

Comment:
pretty sure you can just call `modifiedTestSentence.toArray()` and it will do this automatically